source/backend/api/Services/AcquisitionFileService.cs

```diff
@@ -206,6 +219,33 @@ public PimsAcquisitionFile UpdateProperties(PimsAcquisitionFile acquisitionFile)
             return _acqFileRepository.GetById(acquisitionFile.Internal_Id);
         }
 
+        public PimsAcquisitionFile UpdateChecklistItems(PimsAcquisitionFile acquisitionFile)
+        {
+            acquisitionFile.ThrowIfNull(nameof(acquisitionFile));
+            _logger.LogInformation("Updating acquisition file checklist with AcquisitionFile id: {id}", acquisitionFile.Internal_Id);
+            _user.ThrowIfNotAuthorized(Permissions.AcquisitionFileEdit);
+
+            // Get the current checklist items for this acquisition file.
+            var currentItems = _checklistRepository.GetChecklistItemsByAcquisitionFileId(acquisitionFile.Internal_Id).ToDictionary(ci => ci.Internal_Id);
+
+            foreach (var incomingItem in acquisitionFile.PimsAcquisitionChecklistItems)
+            {
+                if (!currentItems.TryGetValue(incomingItem.Internal_Id, out var existingItem))
```

Comment:
Does this allow for this requirement to be met? 
**if a new checklist item is added to the database it should show up on the UI for all files that are created after it has been added.**

Reply:
We might have to discuss how that is to be achieved for updates. My interpretation is that the code would ensure that new ACQ files would have the new items. Those existing ones would need a DB ETL to bring them up to date so I didn't code any check on the update to add new entries. But if they want this handled via code then we would need to add more logic/complexity to it

Reply:
Check with the Ba's on this one as well, but the AC seems to be stating that this story should be responsible for supporting that requirement, not a separate ETL story.

Reply:
I will confirm with BAs but reading the bolded statement above, it requires that new ACQ get the new item from DB but does not imply anything about updating old ACQ checklists

Reply:
I take it that way because it says, "show in the UI for the files CREATED after the entry was added to the db", not for files UPDATED